projects/js-packages/jetpack-cli/bin/jp.js

```diff
@@ -0,0 +1,286 @@
+
+		console.log( '3. jp docker install' );
+	} catch ( error ) {
+		throw new Error( `Failed to initialize Jetpack: ${ error.message }` );
+	}
+};
+
+// Main execution
+const main = async () => {
+	try {
+		const args = process.argv.slice( 2 );
+
+		// Handle version flag
+		if ( args[ 0 ] === '--version' || args[ 0 ] === '-v' ) {
+			console.log( chalk.green( packageJson.version ) );
+			return;
+		}
+
+		// Handle 'init' command specially
+		if ( args[ 0 ] === 'init' ) {
+			await initJetpack();
+			return;
+		}
+
+		// Try to find monorepo root from current directory
+		const monorepoRoot = findMonorepoRoot( process.cwd() );
+
+		if ( ! monorepoRoot ) {
+			console.error( chalk.red( 'Could not find Jetpack monorepo.' ) );
+
+			console.log( '\nTo get started:' );
+
+			console.log( '1. Run', chalk.blue( 'jp init' ), 'to clone the repository' );
+
+			console.log( '   OR' );
+
+			console.log( '2. Navigate to an existing Jetpack monorepo directory' );
+			throw new Error( 'Monorepo not found' );
+		}
+
+		// Handle docker commands on the host
+		if ( args[ 0 ] === 'docker' ) {
+			// Commands that should run in the container
+			const containerCommands = [ 'build-image', 'install' ];
+			if ( containerCommands.includes( args[ 1 ] ) ) {
+				const result = spawnSync(
+					resolve( monorepoRoot, 'tools/docker/bin/monorepo' ),
+					[ 'pnpm', 'jetpack', ...args ],
+					{
+						stdio: 'inherit',
+						shell: true,
+						cwd: monorepoRoot,
+					}
+				);
+
+				if ( result.status !== 0 ) {
+					throw new Error( `Command failed with status ${ result.status }` );
+				}
+				return;
+			}
+
+			// Run config generation first if this is an 'up' command
+			if ( args[ 1 ] === 'up' ) {
+				// Create required directories
+				fs.mkdirSync( resolve( monorepoRoot, 'tools/docker/data/jetpack_dev_mysql' ), {
+					recursive: true,
+				} );
+				fs.mkdirSync( resolve( monorepoRoot, 'tools/docker/data/ssh.keys' ), { recursive: true } );
+				fs.mkdirSync( resolve( monorepoRoot, 'tools/docker/wordpress' ), { recursive: true } );
+
+				const images = [
+					{ name: 'mariadb:lts' },
+					{ name: 'automattic/jetpack-wordpress-dev:latest' },
+					{ name: 'phpmyadmin/phpmyadmin:latest', platform: 'linux/amd64' },
+					{ name: 'maildev/maildev', platform: 'linux/amd64' },
+					{ name: 'atmoz/sftp', platform: 'linux/amd64' },
+				];
+
+				for ( const image of images ) {
+					const inspect = spawnSync( 'docker', [ 'image', 'inspect', image.name ], {
+						stdio: 'ignore',
+					} );
+					if ( inspect.status !== 0 ) {
+						console.log( chalk.blue( `Pulling ${ image.name }...` ) );
+						const pullArgs = [ 'pull', image.name ];
+						if ( image.platform ) {
+							pullArgs.splice( 1, 0, '--platform', image.platform );
+						}
+						const pull = spawnSync( 'docker', pullArgs, { stdio: 'inherit' } );
+						if ( pull.status !== 0 ) {
+							throw new Error( `Failed to pull ${ image.name }` );
+						}
+					}
+				}
```

Comment:
Doesn't Docker automatically pull images when you don't already have a copy?

Reply:
it does, but it was timing out for me when it attempted to run it during the normal course. I went with just pre-pulling vs rabbit-holing to figure out what was happening. Fair thing to clean up.

Reply:
Now that I'm trying to dive into a rabbit hole, I can't make it fail like it was. Going to remove the prepull logic for now to avoid having to keep up with it. [3f5a875](https://github.com/Automattic/jetpack/pull/40934/commits/3f5a8756c2d80a0fc188d6fa21a398753f217086)